### PR TITLE
Basic avatar system

### PR DESCRIPTION
**Pull request by memetrollsXD**

### CODE REVIEW

**Review by TheLostTree — COMMENTED** [review]
*sigh* the world is your oyster so ig it works

**Review by TheLostTree — APPROVED** [review]

File: src/db/Avatar.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export default class Avatar {
+    private constructor(public ownerUid: UID, public data: AvatarI) {
+
+    }
+
+    public static async create(uid: UID, baseAvatarId: number = 1001) {
+        const db = Database.getInstance();
+        const avatar = new Avatar(uid, {
+            baseAvatarId,
+            equipmentUniqueId: 20003,
```
[the remark this answers is not in the document]
Comment (memetrollsXD) [reply src/db/Avatar.ts:15]: definitely this

[the remark this answers is not in the document]
Comment (memetrollsXD) [reply src/db/Avatar.ts:15]: not that we have inventory implemented yet anyway, so TODO

File: src/server/packets/PlayerLoginCsReq.ts [hunk 2]
```diff
@@ -33,7 +36,40 @@ export default async function handle(session: Session, packet: Packet) {
             stamina: 100,
             worldLevel: 1,
         }
-        plr!.save();
+        plr.save();
+    }
+
+    if (!plr.db.lineup) {
+        Avatar.create(plr.db._id);
```
[the remark this answers is not in the document]
Comment (memetrollsXD) [reply src/server/packets/PlayerLoginCsReq.ts:43]: I added a sanity check if the avatar created already exists anyway. The problem is that the code will later try to fetch the avatar by ID from the lineup (id 1001 which is default for avatar.create)

**Pull request merged**